flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestQueue.java

```diff
@@ -94,10 +96,11 @@ void notifyReaderNonEmpty(final NetworkSequenceViewReader reader) {
 	 * <p>NOTE: Only one thread would trigger the actual enqueue after checking the reader's
 	 * availability, so there is no race condition here.
 	 */
-	private void enqueueAvailableReader(final NetworkSequenceViewReader reader) throws Exception {
-		if (reader.isRegisteredAsAvailable() || !reader.isAvailable()) {
+	private void enqueueAvailableReader(final NetworkSequenceViewReader reader, boolean announceBacklog) throws Exception {
+		if (reader.isRegisteredAsAvailable() || (!reader.isAvailable() && !announceBacklog)) {
```

Comment:
Based on this [comment](https://github.com/apache/flink/pull/11877#discussion_r424353439), we might not need to bring `announceBacklog` argument in this method, to understand all the related processes together.